### ISSUE TITLE
env from next.config.js is undefined in middleware when using turbo in dev.

### ISSUE THREAD

**Issue by qpre** [vercel/next.js#51493]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Mon Apr 24 20:52:24 PDT 2023; root:xnu-8796.121.2~5/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.13.0
      npm: 8.19.3
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.4.7-canary.1
      eslint-config-next: 12.3.4
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/qpre/nextjs-reproduction-turbo-env-middleware

### To Reproduce

run `npm dev` from the reproduction repository.
look at the logs when opening the page.

### Describe the Bug

When setting an env variable in next.config.js.

```
/** @type {import("next").NextConfig} */
module.exports = {
  reactStrictMode: true,

  env: {
    NEXT_PUBLIC_FOO: 'bar',
  },
};
```

and using it in middleware.js 

```
export function middleware(request) {
  const response = NextResponse.next();

  console.log('FOO:', process.env.NEXT_PUBLIC_FOO);

  return response;
}
```

with `next dev --turbo`.

if you look at your logs you'll get:

```
event - [project]/middleware.js changed, and GET / 357ms
FOO: undefined
```

note: I've seen this [PR](https://github.com/vercel/next.js/pull/47767) pass by, it did fix passing the shell's env, but not the one defined from config file.

### Expected Behavior

`process.env.NEXT_PUBLIC_FOO` should be defined in the middleware and the console should log `FOO: bar`.

<sub>[WEB-1209](https://linear.app/vercel/issue/WEB-1209/env-from-nextconfigjs-is-undefined-in-middleware-when-using-turbo-in)</sub>

**Comment by balazsorban44** [comment]
Thanks, I'll pass this as feedback to the team. Keep in mind that Turbopack is still in development, so you can come across some unsupported features like this.

Setting environment variables via `.env*` is the preferred way since Next.js 9.4 anyway, which provides a more intuitive/secure way. You can check out the docs: https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables

**Comment by padmaia** [comment]
This is fixed with v13.5+
